api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -93,11 +93,39 @@ public Schema(int schemaId, List<NestedField> columns, Map<String, Integer> alia
     this.schemaId = schemaId;
     this.struct = StructType.of(columns);
     this.aliasToId = aliases != null ? ImmutableBiMap.copyOf(aliases) : null;
+
+    // validate IdentifierField
+    if (identifierFieldIds != null) {
+      Map<Integer, Integer> idToParent = TypeUtil.indexParents(this.struct);
```

Comment:
Nit: accessing a field doesn't require the prefix `this.` Only assigning a field should use `this.` so that it is clear that the assignment affects the object's state and not just the local state.

Reply:
removed